src/lib/merkle_ledger/util.ml

```diff
@@ -0,0 +1,47 @@
+open Core
+
+module Make (Location : sig
+  type t
+end) (Account : sig
+  type t
+end)
+(Addr : Merkle_address.S)
+(Base : sig
+          type t
+
+          type location
+
+          type addr
+
+          type account
+
+          val get : t -> location -> account option
+
+          val addr_to_location : addr -> location
```

Comment:
When we map a type A to another type B, we usually name it as `B_of_A` (i.e `sexp_of_t`). We do not name it `addr_to_location`. So, this should be named `location_of_addr`.

`location_of_addr` does not fit into `Base`. When you make this new `Inputs_intf`, can you put `location_of_addr` outside of `Base` and it should be directly nested in `Inputs_intf`.

So something like the following:

```ocaml
module type Inputs_intf: sig

module Base : ...
val addr_to_location : ...
end

```